content/news/2025-12-15-fq2bam/index.md

```diff
@@ -0,0 +1,87 @@
+
+
+## FQ2BAM vs BWA-MEM2 mapping statistics
+
+The alignment files (BAM) from FQ2BAM and BWA-MEM2 tools are analysed using [QualiMap BamQC](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fqualimap_bamqc%2Fqualimap_bamqc%2F2.3%2Bgalaxy0&version=latest) tool to compare various mapping statistics on the same human reads. Both alignments are essentially identical in overall mapping performance: they use the same reference size (3,209,286,105 hg38) and number of reads (53,317,838), and both achieve approximately 99.6% mapped reads (53,102,797 for FQ2BAM vs 53,102,800 for BWA-MEM2) with only approximately 0.4% unmapped. Paired-end behavior is also nearly the same (approximately 99.24% “both in pair”) and supplementary alignments are comparable (approximately 0.09%).
+
+<div align="center">
+    <img src="parabricks_qm.png" alt="Runtime comparison of FQ2BAM with BWA-MEM2" width="600"/>
+</div>
+
+FQ2BAM reports 11.2M duplicate reads flagged as duplicates (21.01%) and "duplicated reads skipped" consistent with duplicates being explicitly marked and handled in that pipeline, while the BWA-MEM2 report shows 0 flagged duplicates but a large estimated duplicate count (33.7M; duplication rate 46.16%), suggesting duplicates were not marked and QualiMap BamQC inferred them instead. BWA-MEM2 also has more overlapping read pairs (26.08% vs 20.36%) and slightly more clipped reads (1.29% vs 1.07%), indicating small differences in alignment/fragment handling even though the headline mapping rates are the same.
+
+
+<div align="center">
+    <img src="bwa-mem2_qm.png" alt="Runtime comparison of FQ2BAM with BWA-MEM2" width="600"/>
+</div>
+
+
+### FQ2BAM vs BWA-MEM2 mapping quality comparison
+
+In both "mapping quality across the reference" plots, FQ2BAM and BWA-MEM2 show a very similar overall pattern: the mapping quality stays close to the maximum (around ~60) across most of the genome, indicating that the majority of reads are placed confidently. In both cases, there are sharp, localized drops to low/near-zero mapping quality at specific genomic intervals—typical of repetitive or hard-to-map regions where reads become multi-mapping reads.
+
+<div align="center">
+    <img src="parabricks_mq.png" alt="Mapping quality of FQ2BAM" width="600"/>
+</div>
+
+FQ2BAM appears slightly more uniform overall (image above) but shows a prominent dense block toward the end of the reference with a stronger/extended low-MAPQ signal, whereas BWA-MEM2 shows a more evenly distributed set of chromosome blocks with more frequent narrow MAPQ dips across many chromosomes (image below). Net outcome: both pipelines deliver broadly comparable mapping confidence, with differences mainly in where (and how strongly) low-confidence alignments accumulate.
```

Comment:
> but shows a prominent dense block toward the end of the reference

compare the sort order of the contigs reported by qualimap for the bwa-mem2 vs parabricks output.
parabricks tends to lump the _alt contigs, etc. together near the end, while bwa-mem2 sorts alphabetically. All these small contigs produce "dense block" in the plot.

In other words there is not necessarily any alignment difference here, but a visual artifact.

Reply:
I sorted (samtools sort by coordinate) the BAM file produced by Parabricks but it produces the same mapping quality plot using QualiMap BamQC

<img width="787" height="643" alt="samtools_sort" src="https://github.com/user-attachments/assets/5e46c244-2c34-4e4f-85f1-f6213fa23a8f" />